ibis/__init__.py

```diff
@@ -3,18 +3,22 @@
 from contextlib import suppress
 
 import ibis.config_init  # noqa: F401
-import ibis.expr.api as api  # noqa: F401
-import ibis.expr.types as ir  # noqa: F401
-import ibis.util as util  # noqa: F401
+import ibis.expr.types as ir
+from ibis import util
 
 # pandas backend is mandatory
 from ibis.backends import pandas  # noqa: F401
-from ibis.common.exceptions import IbisError  # noqa: F401
-from ibis.config import options  # noqa: F401
+from ibis.common.exceptions import IbisError
+from ibis.config import options
+from ibis.expr import api
 from ibis.expr.api import *  # noqa: F401,F403
 
 from ._version import get_versions  # noqa: E402
 
+__all__ = ['api', 'ir', 'util', 'IbisError', 'options']
+__all__ += api.__all__
```

Comment:
I guess you mean the `api.__all__`. Since we have the `from ibis.expr.api import *` everything in the `api` namespace is also available in the `ibis` namespace. If I remove this line, I'll change the behaviour of the module, and `dir(ibis)` won't include those.

Personally I don't think this duplication of the API is great. Making it explicit is not a way to acknowledge that this is what we want, as much as a way to make it visible what we are doing, so it's clearer and also we can better think if we should change this.

I'm happy to remove that line, but I don't think we should change the behaviour in a refactoring PR. I think making things explicit here is better, and then follow up. Also, worth noting that the main reason for this PR is that this is creating trouble in the entrypoints PR, since I'm using `__getattr__` to access the backends, and this clutter is creating some issues.

I added the test, let me know if it looks good now, or if you still want me to remove that line.

Reply:
yeah we prob want to make this simpler over time, but preserving for now is good.